app/models/package_manager/nu_get.rb

```diff
@@ -70,8 +76,40 @@ def self.project(name)
       h
     end
 
+    def self.package_file(name, version)
+      url = download_url_by_name(name, version)
+      StringIO.new(get_raw(url))
+    end
+
+    def self.nuspec(name, version)
+      nuspec = nil
+
+      Zip::File.open_buffer(package_file(name, version)) do |zip|
+        zip.each do |file|
+          # the correct nuspec file will be at the root of the tree with the package name in the filename
+          # package identifiers are case-insensitive: https://learn.microsoft.com/en-us/nuget/reference/nuspec#id
+          file.get_input_stream { |io| nuspec = Ox.parse(io.read) } if file.name.downcase["#{name}.nuspec".downcase]
+        end
+      end
+
+      Rails.logger.error(
```

Comment:
Should there be an if statement condition that needs to be met before logging this?

Reply:
D'oh, yes.